contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,445 @@
+    return <Text>{foo}</Text>;
+  }
+
+  // bad
+  import { ComponentProps } from "React";
+  import MyComponent from "./MyComponent";
+  type MyComponentProps = ComponentProps<typeof MyComponent>;
+
+  // good
+  import MyComponent, { MyComponentProps } from "./MyComponent";
+  ```
+
+<a name="convension-file-organization"></a><a name="1.15"></a>
+
+- [1.15](#convension-file-organization) **File organization**: In modules with platform-specific implementations, create `types.ts` to define shared types. Import and use shared types in each platform specific files.
+
+  > Why? To encourage consistent API across platform-specific implementations.
+
+  Utility module example
+
+  ```ts
+  // types.ts
+  type GreetingModule = {
+    getHello: () => string;
+    getGoodbye: () => string;
+  };
+
+  // index.native.ts
+  import { GreetingModule } from "./types.ts";
+  function getHello() {
+    return "hello from mobile code";
+  }
+  function getGoodbye() {
+    return "goodbye from mobile code";
+  }
+  const Greeting: GreetingModule = {
+    getHello,
+    getGoodbye,
+  };
+  export default Greeting;
+
+  // index.ts
+  import { GreetingModule } from "./types.ts";
+  function getHello() {
+    return "hello from other platform code";
+  }
+  function getGoodbye() {
+    return "goodbye from other platform code";
+  }
+  const Greeting: GreetingModule = {
+    getHello,
+    getGoodbye,
+  };
+  export default Greeting;
+  ```
+
+  Component module example
+
+  ```ts
+    // types.ts
+    export type MyComponentProps = {
+      foo: string;
+    }
+
+    // index.ios.ts
+    import { MyComponentProps } from ./types.ts;
+
+    export MyComponentProps;
+    export default function MyComponent({ foo }: MyComponentProps) {...}
+
+    // index.ts
+    import { MyComponentProps } from ./types.ts;
+
+    export MyComponentProps;
+    export default function MyComponent({ foo }: MyComponentProps) {...}
+  ```
+
+<a name="reusable-types"></a><a name="1.16"></a>
+
+- [1.16] **Reusable Types**: Reusable type definitions, such as models (e.g. Report), must have their own file and be placed under `src/types/`. The type should be exported as a default export.
+
+  ```ts
+  // src/types/Report.ts
+
+  type Report = {...};
+
+  export default Report;
+  ```
+
+## Communication Items
+
+> Comment in the `#expensify-open-source` Slack channel if any of the following situations are encountered. Each comment should be prefixed with `TS ATTENTION:`. Internal engineers will access each situation and prescribe solutions to each case. Internal engineers should refer to general solutions to each situation that follows each list item.
+
+- I think types definitions in a third party library is incomplete or incorrect
+
+When the library indeed contains incorrect type definitions and it cannot be updated, use module augmentation to correct them. All module augmentation code should be contained in `global.d.ts`.
```

Comment:
```suggestion
When the library indeed contains incorrect type definitions and it cannot be updated, use module augmentation to correct them. All module augmentation code should be contained in `/src/global.d.ts`.
```

☝️ To be more specific to put it in one `global.d.ts`. cc @hayata-suenaga @fabioh8010

Reply:
nice catch! 👍